src/components/VisibilityContainer/index.tsx

```diff
@@ -0,0 +1,23 @@
+import { useRef, useState } from "react";
+import useIntersectionObserver from "hooks/useIntersectionObserver";
+
+interface VisibilityContainerProps {
+  children: React.ReactNode;
+}
+/**
+ * `VisibilityContainer` is a component that will only render its children when it is visible in the viewport.
```

Comment:
I'm not sure how much this would differ in production, but if there's a lot of content,  there's a noticeable lag between when it actually loads in. Is that expected? (this would probably only affect large projects, but I only checked on dev since beta is occupied right now)

https://user-images.githubusercontent.com/47064971/209198264-7d9e3d94-0bd6-43ea-bdbe-44a37be4c643.mov

Reply:
This is just a result of rendering so many icons. Currently, this lag exists but it all happens on the initial render as opposed to when scrolling. In most scenarios, it shouldn't be noticeable unless they open a build variant with hundreds of task icons.